packages/plugins/opentelemetry/src/plugin.ts

```diff
@@ -0,0 +1,291 @@
+          },
+          rootContext,
+        );
+
+        return ({ result }) => {
+          if (result instanceof Error) {
+            parseSpan.setStatus({
+              code: SpanStatusCode.ERROR,
+              message: result.message,
+            });
+          }
+
+          parseSpan.end();
+        };
+      }
+    },
+    onExecute({ args }) {
+      const shouldTraceExecute = options.spans?.graphqlExecute ?? true;
+      if (requestContextMapping.has(args.contextValue.request)) {
+        const rootContext = requestContextMapping.get(args.contextValue.request)!;
+        const operation = getOperationAST(args.document, args.operationName);
+        const executeSpan = tracer.startSpan(
+          'graphql.execute',
+          {
+            attributes: {
+              [SEMATTRS_GRAPHQL_OPERATION_TYPE]: operation.operation,
+              [SEMATTRS_GRAPHQL_OPERATION_NAME]: args.operationName,
+              [SEMATTRS_GRAPHQL_DOCUMENT]: print(args.document),
+            },
+            kind: SpanKind.INTERNAL,
+          },
+          rootContext,
+        );
+
+        return {
+          onExecuteDone({ result }) {
+            if (!isAsyncIterable(result)) {
+              if (result.errors && result.errors.length > 0) {
+                executeSpan.setAttribute(SEMATTRS_GRAPHQL_ERROR_COUNT, result.errors.length);
+                executeSpan.setStatus({
+                  code: SpanStatusCode.ERROR,
+                  message: result.errors.map(e => e.message).join(', '),
+                });
+
+                for (const error in result.errors) {
+                  executeSpan.recordException(error);
+                }
+              }
+            }
+
+            executeSpan.end();
+          },
+        };
+      }
+    },
+    onFetch({ context, options, url, setOptions, executionRequest }) {
+      if (requestContextMapping.has(context.request)) {
+        const otelContext = requestContextMapping.get(context.request)!;
+        const reqHeaders = getHeadersObj(options.headers || {});
+        propagation.inject(otelContext, reqHeaders);
+
+        setOptions({
+          ...options,
+          headers: reqHeaders,
+        });
+
+        const urlObj = new URL(url);
+        const fetchSpan = tracer.startSpan(
+          'http.fetch',
+          {
+            attributes: {
+              [SEMATTRS_HTTP_METHOD]: options.method,
+              [SEMATTRS_HTTP_URL]: url,
+              [SEMATTRS_NET_HOST_NAME]: urlObj.hostname,
+              [SEMATTRS_HTTP_HOST]: urlObj.host,
+              [SEMATTRS_HTTP_ROUTE]: urlObj.pathname,
+              [SEMATTRS_HTTP_SCHEME]: urlObj.protocol,
+              [SEMATTRS_HTTP_USER_AGENT]: reqHeaders['user-agent'],
+            },
+            kind: SpanKind.CLIENT,
+          },
+          otelContext,
+        );
+
+        if (executionRequest) {
+          fetchSpan.setAttribute(SEMATTRS_GRAPHQL_OPERATION_NAME, executionRequest.operationName);
```

Comment:
Shouldn't this be in `onSubgraphExecute`? Just curious.

Reply:
Not really, because I don't want to create an additional span. I want to just "enrich" the upstream-fetch-HTTP-span with the subgraph data.
No need to make additional spans in my opinion.